From: "Yan, Zheng" <zyan@redhat.com>
Date: Tue, 19 Sep 2017 14:38:13 +0800
Subject: client: set client_try_dentry_invalidate to false by default

By default, ceph-fuse uses side effect of 'dentry invalidation' to
trim kernel dcache if it runs on kernel < 3.18. The implemention of
kernel function d_invalidate() changed in 3.18 kernel, the method no
longer works for upstream kernel >= 3.18.

RHEL 3.10 kernel includes backport of patches that change implemention
of d_invalidate(). So checking kernel version to decide if 'dentry
invalidation' method works is unreliable.

Fixes: http://tracker.ceph.com/issues/21423
Signed-off-by: "Yan, Zheng" <zyan@redhat.com>
(cherry picked from commit 69adaabface27880dd6c8dbfdeeb06cf3f3d346a)

Conflicts:
    src/common/options.cc
(cherry picked from commit 98b357e68add714adac1d9902ad1da5ef368b063)

Resolves: rhbz#1492977
---
 src/common/config_opts.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/config_opts.h b/src/common/config_opts.h
index cef6a05..42209b3 100644
--- a/src/common/config_opts.h
+++ b/src/common/config_opts.h
@@ -414,7 +414,7 @@ OPTION(fuse_multithreaded, OPT_BOOL, true)
 OPTION(fuse_require_active_mds, OPT_BOOL, true) // if ceph_fuse requires active mds server
 OPTION(fuse_syncfs_on_mksnap, OPT_BOOL, true)
 
-OPTION(client_try_dentry_invalidate, OPT_BOOL, true) // the client should try to use dentry invaldation instead of remounting, on kernels it believes that will work for
+OPTION(client_try_dentry_invalidate, OPT_BOOL, false) // the client should try to use dentry invaldation instead of remounting, on kernels it believes that will work for
 OPTION(client_die_on_failed_remount, OPT_BOOL, true)
 OPTION(client_check_pool_perm, OPT_BOOL, true)
 OPTION(client_use_faked_inos, OPT_BOOL, false)
